implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AbstractKeyLocationResolver.java

```diff
@@ -85,17 +91,31 @@ protected static String getKid(JsonWebStructure jws) {
         return jws.getHeaders().getStringHeaderValue(JsonWebKey.KEY_ID_PARAMETER);
     }
 
-    protected HttpsJwks initializeHttpsJwks(String location) throws IOException {
+    protected HttpsJwks initializeHttpsJwks(String location, String tlsCertificatePath, Set<String> trustedHosts)
+            throws IOException {
         PrincipalLogging.log.tryCreateKeyFromHttpsJWKS();
         HttpsJwks theHttpsJwks = new HttpsJwks(location);
         theHttpsJwks.setDefaultCacheDuration(authContextInfo.getJwksRefreshInterval().longValue() * 60L);
+        if (location.startsWith(HTTPS_SCHEME)) {
+            Get httpGet = new Get();
+            if (tlsCertificatePath != null) {
+                httpGet.setTrustedCertificates(loadPEMCertificate(readKeyContent(tlsCertificatePath)));
+            }
+            if (trustedHosts != null) {
+                httpGet.setHostnameVerifier(new TrustedHostsHostnameVerifier(trustedHosts));
+            } else {
+                httpGet.setHostnameVerifier(new TrustAllHostnameVerifier());
+            }
```

Comment:
Should the default be more defensive than trust all? I'm not sure :-)

Reply:
Hi @MikeEdgar Yeah, good point, I think I'll need to introduce one more boolean property, `smallrye.jwt.tls.trust-all`, `false` is default

Reply:
Mike, @MikeEdgar It should be better now, have a look please. Perhaps it can make sense for you to verify that the workaround you had to use is no longer necessary with this PR ?

Reply:
I would expect this to verify hostnames by default but not require a restricted set of hostnames, unless trust-all is enabled.

Reply:
> Perhaps it can make sense for you to verify that the workaround you had to use is no longer necessary with this PR ?

LGTM - I'll try it as a replacement for the workaround probably early next week.

Reply:
Thanks @MikeEdgar